### PR TITLE
RFCS: update status fields and clean up issue/PR links

### PR DESCRIPTION
**Pull request by a-robinson**
@knz I wasn't sure what the status of https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20170517_algebraic_data_types.md is

### CODE REVIEW

File: docs/RFCS/20160730_streaming_snapshots.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 - Feature Name: streaming_snapshots
-- Status: in-progress
+- Status: completed
```
Comment (jordanlewis) [inline docs/RFCS/20160730_streaming_snapshots.md]: I think this might be still in-progress - the receiver doesn't stream the snapshot yet.

Comment (a-robinson) [reply docs/RFCS/20160730_streaming_snapshots.md]: You're right, thanks. Changed back.

**Comment by a-robinson** [comment]
@petermattis or anyone really can I get an LGTM?

**Review by benesch — APPROVED** [review]

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 26 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[docs/RFCS/20170602_distsql_limit.md, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-L-YQjG5fPGijI2AUBcz:-L-YQjG5fPGijI2AUBd-:b-ws2fbg) ([raw file](https://github.com/cockroachdb/cockroach/blob/066e380339c9941f28bfdd316d0167383cb9ea3a/docs/RFCS/20170602_distsql_limit.md#L2)):*
> ```Markdown
> - Feature Name: Limit handling in DistSQL
> - Status: completed
> ```

@raduberinde is this complete?

---

*[docs/RFCS/20170605_dedicated_raft_storage.md, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-L-YQoQbmsEwmSOrvB1q:-L-YQoQbmsEwmSOrvB1r:bgpfw5t) ([raw file](https://github.com/cockroachdb/cockroach/blob/066e380339c9941f28bfdd316d0167383cb9ea3a/docs/RFCS/20170605_dedicated_raft_storage.md#L2)):*
> ```Markdown
> - Feature Name: Dedicated storage engine for Raft
> - Status: in-progress
> ```

I think this can be `postponed`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-:-L-YQzRKwZ73wV0y6Gt0:bnfp4nl)*

**Comment by RaduBerinde** [comment]
I updated a couple in #20453

---

Review status: 0 of 26 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[docs/RFCS/20170602_distsql_limit.md, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-L-YQjG5fPGijI2AUBcz:-L-YTOXgS31qEHNZbG2O:b9i69th) ([raw file](https://github.com/cockroachdb/cockroach/blob/066e380339c9941f28bfdd316d0167383cb9ea3a/docs/RFCS/20170602_distsql_limit.md#L2)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

@raduberinde is this complete?
</blockquote></details>

No. It's not even started.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-:-L-YTGVK4QTrw9knFkFb:b-kwwhn8)*

**Comment by a-robinson** [comment]
Review status: 0 of 23 files reviewed at latest revision, 3 unresolved discussions.

---

*[docs/RFCS/20170602_distsql_limit.md, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-L-YQjG5fPGijI2AUBcz:-L-YY-5msMxn3I4muaQ-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/066e380339c9941f28bfdd316d0167383cb9ea3a/docs/RFCS/20170602_distsql_limit.md#L2)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

No. It's not even started.
</blockquote></details>

Done.

---

*[docs/RFCS/20170605_dedicated_raft_storage.md, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455#-L-YQoQbmsEwmSOrvB1q:-L-YXyeX3NBIcKL4V6zQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/066e380339c9941f28bfdd316d0167383cb9ea3a/docs/RFCS/20170605_dedicated_raft_storage.md#L2)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I think this can be `postponed`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20455)*

**Pull request merged**